ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -508,6 +520,181 @@ const triggerSet: TriggerSet<Data> = {
         },
       },
     },
+    {
+      id: 'P12S Engravement of Souls Tracker',
+      type: 'Ability',
+      netRegex: { id: '8305', source: 'Athena', capture: false },
+      run: (data) => ++data.engravementCounter,
+    },
+    // In Engravement 1 (Paradeigma 2), 2 players receive lightTower and 2 players receive darkTower,
+    // 2 players need to guide the light beam and 2 players need to guide the dark beam.
+    // The operator of the beam extends the beam directly from the outside. The beam is attenuated until the jagged line disappears.
+    // The people in the tower find the people who have the opposite attribute to the debuff and put them in four places.
+    // At NE NW SE SW as a # shape. The position of outside Anthropos is fixed by two situation.
+    // {[97, 75], [125, 97], [103, 125], [75, 103]} and {[103, 75], [125, 103], [97, 125], [75, 97]}. The Anthropos will cast
+    // 'Searing Radiance' for light beam and 'Shadowsear' for dark beam. We use those as a trigger for Tower players place
+    // the Tower.
+    // When debuffs expire and towers drop, their debuff changes to lightTilt or darkTilt (same as tower color).
+    // At the same time the towers drop, the 4 tethered players receive lightTilt or darkTilt depending on their tether color.
+    //
+    {
+      id: 'P12S Engravement 1 Drop Tower Pos Tracker',
+      type: 'StartsUsing',
+      netRegex: { id: ['82F[12]'], source: 'Anthropos' },
+      condition: (data) => data.engravementCounter === 1,
+      promise: async (data, matches) => {
+        data.combatantData = [];
+        const ids = parseInt(matches.sourceId, 16);
+        if (ids === undefined)
+          return;
+        data.combatantData = (await callOverlayHandler({
+          call: 'getCombatants',
+          ids: [ids],
+        })).combatants;
+      },
+      run: (data, matches) => {
+        const x = data.combatantData[0]?.PosX;
+        if (x === undefined)
+          return;
+        const y = data.combatantData[0]?.PosY;
+        if (y === undefined)
+          return;
+
+        if (x === 75 && y === 97) {
+          if (matches.id === '82F1')
+            data.engravement1BeamsPosMap.set('NE', 'dark');
+          else
+            data.engravement1BeamsPosMap.set('NE', 'light');
```

Comment:
Logic simplification request: I think you could set `const color = matches.id === '82F1' ? 'dark' : 'light'` and then simplify a lot of this logic by setting things in the map to `color` instead of needing if statements.